From 5757dc120824d4aa1d08fce0a48b9f9d6bc68311 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <xiaowen.liu@freescale.com>
Date: Fri, 19 Jun 2015 17:45:59 +0800
Subject: [PATCH 586/635] MA-6784 workaround cache memory allocation.

When CMA memory is consumed up by system, GPU allocate cache memory
from CMA will cost time. Disable cache memory allocation to workaround
this issue to improve gralloc memory allocation performance.

Signed-off-by: Liu Xiaowen <xiaowen.liu@freescale.com>
---
 mx6/libgralloc_wrapper/BufferManager.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mx6/libgralloc_wrapper/BufferManager.cpp b/mx6/libgralloc_wrapper/BufferManager.cpp
index e0aa96b..70f4d21 100755
--- a/mx6/libgralloc_wrapper/BufferManager.cpp
+++ b/mx6/libgralloc_wrapper/BufferManager.cpp
@@ -162,6 +162,11 @@ bool BufferManager::useFSLGralloc(int format, int usage)
 {
     bool bUseFSLGralloc = true;
 
+    if (((usage & GRALLOC_USAGE_SW_WRITE_OFTEN) == GRALLOC_USAGE_SW_WRITE_OFTEN) ||
+            ((usage & GRALLOC_USAGE_SW_READ_OFTEN) == GRALLOC_USAGE_SW_READ_OFTEN)) {
+        return true;
+    }
+
     //RGB format and without video encoder flag
     if ((format >= HAL_PIXEL_FORMAT_RGBA_8888 && format <= HAL_PIXEL_FORMAT_sRGB_X_8888)
        && !(usage & GRALLOC_USAGE_HW_VIDEO_ENCODER)) {
-- 
1.8.0

